src/py/reactpy/reactpy/core/state_recovery.py

```diff
@@ -0,0 +1,295 @@
+import asyncio
+import base64
+from dataclasses import asdict, is_dataclass
+import datetime
+import hashlib
+import time
+from collections.abc import Iterable
+from decimal import Decimal
+from logging import getLogger
+from pathlib import Path
+from typing import Any, Callable
+from uuid import UUID
+
+import orjson
+import pyotp
+from more_itertools import chunked
+
+logger = getLogger(__name__)
+
+
+class StateRecoveryFailureError(Exception):
+    """
+    Raised when state recovery fails.
+    """
+
+
+class StateRecoveryManager:
+    def __init__(
+        self,
+        serializable_types: Iterable[type],
+        pepper: str,
+        otp_key: str | None = None,
+        otp_interval: int = (4 * 60 * 60),
+        otp_digits: int = 10,  # 10 is the max allowed
+        otp_max_age: int = (48 * 60 * 60),
+        # OTP code is actually three codes, in the past and future concatenated
+        otp_mixer: float = (365 * 24 * 60 * 60 * 3),
+        max_num_state_objects: int = 512,
+        max_object_length: int = 40000,
+        default_serializer: Callable[[Any], bytes] | None = None,
```

Comment:
This is slightly misleading. When serializing and an object not supported by `orjson` is encountered, it falls back to a default. However, a default is already provided for types like timezone, timedelta, and dataclasses. If the provided default cannot serialize the object, then this is the fallback for that.